front/components/providers/AI21Setup.jsx

```diff
@@ -13,9 +13,11 @@ export default function AI21Setup({ open, setOpen, config, enabled }) {
   const [testError, setTestError] = useState("");
   const [enableRunning, setEnableRunning] = useState(false);
 
-  if (config && config.api_key.length > 0 && apiKey.length == 0) {
-    setApiKey(config.api_key);
-  }
+  useEffect(() => {
```

Comment:
The react linter is still unhappy.
You can re-enable it by deleting these lines temporarily 
<img width="1148" alt="Screen Shot 2023-04-04 at 10 49 01" src="https://user-images.githubusercontent.com/358965/229739610-0ec42bad-2896-4a53-ad0b-e946b78be03c.png">

and you will see the errors messages I am referring to:
![Uploading Screen Shot 2023-04-04 at 10.50.42.png…]()

Reply:
Once you are done fixing the errors for this PR, you can just put back these lines and I will take care of the code base clean up.